### PR TITLE
Request

### PR DESCRIPTION
**Pull request by kimchhunnn**

### CODE REVIEW

File: Helper/HttpRequest.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                //Send the GET request
+                httpResponse = await httpClient.GetAsync(requestUri);
+                httpResponseBody = await httpResponse.Content.ReadAsStringAsync();
+
+            }
+            catch (Exception ex)
+            {
+
+                string aa = "Error: " + ex.HResult.ToString("X") + " Message: " + ex.Message;
+            }
```
Comment (kenphanith) [inline Helper/HttpRequest.cs]: @kimchhunnn This is working ?

File: MainPage.xaml [hunk 2]
```diff
@@ -122,10 +125,10 @@
                                     Margin="5,0,0,3"
                                     Grid.Column="1"
                                     VerticalAlignment="Bottom"
-                                    Text="{x:Bind FullName}" />
+                                    Text="{Binding FullName}" />
```
Comment (kenphanith) [inline MainPage.xaml]: @kimchhunnn Why did you change from x:Bind to Binding ?

File: MainPage.xaml.cs [hunk 3]
```diff
@@ -22,81 +27,67 @@ public sealed partial class MainPage : Page
...
-
-            var headers = httpClient.DefaultRequestHeaders;
-            string header = "ie";
-            if (!headers.UserAgent.TryParseAdd(header))
+            string uri = "";
+            if (this.Keyword.Text.Length > 0)
             {
-                throw new Exception("Invalid header value: " + header);
+                uri = "https://api.github.com/search/repositories?q=" + this.Keyword.Text;
             }
```
Comment (kenphanith) [inline MainPage.xaml.cs]: @kimchhunnn if your keyword is empty, your request uri is empty. Does it make sense ?

File: MainPage.xaml.cs [hunk 4]
```diff
@@ -22,81 +27,67 @@ public sealed partial class MainPage : Page
...
+                    this.ResultList.ItemsSource = result.Items;
+                    this.ResultMsg.Text = result.Items.Count() + " Results";
+                }
             }
-            catch (Exception ex)
+            catch (Exception err)
             {
-                httpResponseBody = "Error: " + ex.HResult.ToString("X") + " Message: " + ex.Message;
-            }
+                ErrorResponse errorResponse = Newtonsoft.Json.JsonConvert.DeserializeObject<ErrorResponse>(response);
```
Comment (kenphanith) [inline MainPage.xaml.cs]: @kimchhunnn This code is working ?

File: MainPage.xaml (hunk 2 above)
Comment (kimchhunnn) [reply MainPage.xaml]: @kenphanith When I moved Repo class out of MainPage.xaml.cs, x:bind throws an error when run

File: Helper/HttpRequest.cs (hunk 1 above)
Comment (kimchhunnn) [reply Helper/HttpRequest.cs]: @kenphanith Fixed

File: MainPage.xaml.cs (hunk 3 above)
Comment (kimchhunnn) [reply MainPage.xaml.cs]: @kenphanith When the keyword is empty, it will go to else case and return void. No further execution.

File: MainPage.xaml (hunk 2 above)
Comment (kenphanith) [reply MainPage.xaml]: @kimchhunnn Do you know the different between both ?

File: MainPage.xaml.cs (hunk 3 above)
Comment (kenphanith) [reply MainPage.xaml.cs]: @kimchhunnn It make sense ?

File: Helper/HttpRequest.cs [hunk 5]
```diff
@@ -40,8 +40,7 @@ public static async Task<string> HttpGetRequest(string uri)
             }
             catch (Exception ex)
             {
-
-                string aa = "Error: " + ex.HResult.ToString("X") + " Message: " + ex.Message;
+                httpResponseBody = "Error: " + ex.HResult.ToString("X") + " Message: " + ex.Message;
             }
             return httpResponseBody;
```
Comment (kenphanith) [inline Helper/HttpRequest.cs]: @kimchhunnn Is it the right to catch an error ?

File: MainPage.xaml (hunk 2 above)
Comment (kimchhunnn) [reply MainPage.xaml]: @kenphanith Yes, I used Binding because I don't have to create an object and mention the path to the type I want to use.

File: MainPage.xaml.cs (hunk 3 above)
Comment (kimchhunnn) [reply MainPage.xaml.cs]: @kenphanith May I have your idea, please?

File: MainPage.xaml.cs (hunk 4 above)
Comment (kimchhunnn) [reply MainPage.xaml.cs]: @kenphanith It didn't work well. I updated it.

File: Helper/HttpRequest.cs (hunk 5 above)
Comment (kimchhunnn) [reply Helper/HttpRequest.cs]: @kenphanith It didn't work well. I updated it.
